File: 프로그래머스/lv1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        char[] chars = s.toCharArray(); 
        for (int i = 0; i < chars.length; i++) {
            char c = chars[i];
            if (c == ' ') {
                continue;   
            }
            boolean isSmall = c > 'Z';
            c = (char) (c + n);
            
            if ((isSmall && c > 'z') || (!isSmall && c > 'Z')) {
                c = (char) (c - 26);
            }
            chars[i] = c;
        }
        return String.valueOf(chars);
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    let arr = s.split('');
    for (let i = 0; i < arr.length; i++) {
        let alpha = arr[i];
        arr[i] = alpha === ' ' ? alpha : shift(alpha.charCodeAt(), n);
    }
    return arr.join('');
}

function shift(alpha, num) {
    let isSmall = alpha > 90;
    alpha += num;
    if ((isSmall && alpha > 122) || (!isSmall && alpha > 90)){
        alpha -= 26;
    }
    return String.fromCharCode(alpha);
}